File: atCoderEnv/problems/tessoku-book-py/a62/a62.py
from sys import stdin, setrecursionlimit

setrecursionlimit(10**6)


def main():
    N, M = map(int, stdin.readline().split())

    graph = [[] for _ in range(N)]
    for _ in range(M):
        A, B = map(int, stdin.readline().split())
        graph[A-1].append(B-1)
        graph[B-1].append(A-1)

    visited = [False]*N

    def dfs(v):
        visited[v] = True
        for adj in graph[v]:
            if visited[adj] == False:
                dfs(adj)
    dfs(0)

    if False in visited:
        print("The graph is not connected.")
    else:
        print("The graph is connected.")


if __name__ == '__main__':
    main()

File: atCoderEnv/problems/tessoku-book/a62/a62.cpp
#include <bits/stdc++.h>
using namespace std;
#include <atcoder/all>
using namespace atcoder;
/* alias */
using ull = unsigned long long;
using ll = long long;
using vi = vector<int>;
using vvi = vector<vi>;  // 二次元vector
using vs = vector<string>;
using pii = pair<int, int>;
/* macro */
#define rep(i, a, n) for (int i = (a); i < (n); ++i)

int N, M;
vi G[100009];
bool visited[100009];

void dfs(int v) {
  visited[v] = true;
  for (int adj : G[v]) {
    if (visited[adj] == false) {
      dfs(adj);
    }
  }
}

int main() {
  cin >> N >> M;
  rep(i, 1, N + 1) visited[i] = false;
  rep(i, 0, M) {
    int a, b;
    cin >> a >> b;
    G[a].push_back(b);
    G[b].push_back(a);
  }
  dfs(1);
  rep(i, 1, N + 1) {
    if (visited[i] == false) {
      cout << "The graph is not connected." << endl;
      return 0;
    }
  }
  cout << "The graph is connected." << endl;
  return 0;
}